### PR TITLE
dont poke routes if we have no exit

### PR DESCRIPTION
**Pull request by majestrate**
if we have no exit enabled disable the route poking.

### CODE REVIEW

**Comment by majestrate** [comment]
rebased

File: llarp/rpc/rpc_server.cpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #include "rpc_server.hpp"
-#include <router/abstractrouter.hpp>
+#include <router/router.hpp>
```
Comment (tewinget) [inline llarp/rpc/rpc_server.cpp:2]: If things need to be added to a class' interface, that should be done via the interface class.  If that feels cumbersome, probably a sign that kitchen sinking is happening.

**Comment by majestrate** [comment]
squashed and redid commits so they are cleaner.

File: llarp/rpc/rpc_server.cpp (hunk 1 above)
Comment (majestrate) [reply llarp/rpc/rpc_server.cpp]: done, squashed and rebased

**Review by tewinget — APPROVED** [review]
Looks fine.

**Pull request merged**